server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitfinex/BitfinexExchange.java

```diff
@@ -108,9 +114,44 @@ public BitfinexDepth getBitfinexOrderBook(String pair, Integer limitBids, Intege
         protected void initServices() {
             super.initServices();
             this.marketDataService = new HotfixBitfinexMarketDataService(this);
+
+            // WORKAROUND for BCH deposit - remove after released in upstream
+            // https://github.com/knowm/XChange/pull/3507
+            this.accountService = new BitfinexAccountService(this) {
+
+
+                @Override
+                public String requestDepositAddress(Currency currency, String... arguments) throws IOException {
+                    try {
+                        final BitfinexDepositAddressResponse response =
+                            requestDepositAddressRaw(currency.getCurrencyCode());
+                        return response.getAddress();
+                    } catch (BitfinexException e) {
+                        throw BitfinexErrorAdapter.adapt(e);
+                    }
+                }
+
+                @Override
+                public BitfinexDepositAddressResponse requestDepositAddressRaw(String currency) throws IOException {
+                    if (!currency.equalsIgnoreCase("BCH")) {
+                        return super.requestDepositAddressRaw(currency);
+                    }
+
+                    String type = "bab"; // bitcoin cash (ABC)
```

Comment:
different method for BCH, PR created for Xchange library: https://github.com/knowm/XChange/pull/3507